dataset/src/main/scala/frameless/TypeCoercion.scala

```diff
@@ -20,8 +20,15 @@ trait TypeCoercion[LeftType, RightType] {
 }
 
 object TypeCoercion {
+  type Exists[A, B, C] = TypeCoercion[A, B] {type ResultType = C}
 
-  type Aux[A, B, C] = TypeCoercion[A, B] {type ResultType = C}
+  def apply[A,B](implicit tc: TypeCoercion[A, B]): Exists[A,B,tc.ResultType] = tc
+}
+
+/**
+  * The following conversions need to be explicitly brought into scope otherwise only same type operations are allowed.
+  */
+object widenning {
```

Comment:
@OlivierBlanvillain @kanterov now you need to import frameless.widenning._ to get these conversions.